apollographql/apollo-client-devtools
# Get the development app to run

I was having a heck of a time trying to spin up the development app in the `development` folder due to peer dependency clashes, outdated APIs, etc.

This PR aims to modernize some of the development app to bring it up to date. Specifically:

* Upgrade the server to `@apollo/server` v4
* Upgrade the server to use `@apollo/datasource-rest`
* Update the client to use React 18
* Upgrade to the latest `@apollo/client` version
* Upgrade to the latest `@typescript-eslint/eslint-plugin`
* Upgrade to the latest `@typescript-eslint/parser`

This PR also addresses some linting issues:

* Adds the `setting` configuration for ESLint to auto detect the React version
* Allow the `css` prop on React elements
* Fix conditional hook usage in `EntityView`

This app should now run properly when cloning the repo fresh and trying to install it.

---

To test this PR:

1. Clone a fresh copy of the repo

```sh
git clone git@github.com:apollographql/apollo-client-devtools.git
```

2. Install dependencies

```sh
npm install
npm run install:dev
```

3. Start the app

```sh
npm run start:dev
```

You should now be able to navigate to http://localhost:3000 and see the app running.
